pandas/core/dtypes/missing.py

```diff
@@ -315,6 +319,35 @@ def _isna_string_dtype(values: np.ndarray, inf_as_na: bool) -> npt.NDArray[np.bo
     return result
 
 
+def _check_record_value(element: Any, inf_as_na: bool) -> np.bool_:
+    is_element_nan = False
+    if element != element:
+        is_element_nan = True
+
+    is_element_inf = False
+    if inf_as_na:
+        try:
+            if np.isinf(element):
+                is_element_inf = True
+        except TypeError:
+            is_element_inf = False
+
+    return np.any(np.logical_or(is_element_nan, is_element_inf))
+
+
+def _isna_recarray_dtype(values: np.recarray, inf_as_na: bool) -> npt.NDArray[np.bool_]:
+    result = np.zeros(values.shape, dtype=bool)
+    for i, record in enumerate(values):
+        does_record_contain_nan = np.zeros(len(record), dtype=bool)
+        for j, element in enumerate(record):
```

Comment:
Can you just call `isnan` and/or `isinf` on `record`?

Reply:
@mroeschke I used isnan_all for the record and I rewrote a little bit the isinf part but I cannot use `isinf`directly on the record since it will throw a `TypeError` when the record value are strings. But I hope the way I rewrote the method still improved on the previous version.